### PR TITLE
add warnings when exceeding license claims

### PR DESCRIPTION
**Pull request by wdbaruni**
## Overview
This PR introduces a license validation system for Bacalhau that checks for license validity and node limits, and communicates status to users through friendly warnings.

## Key Features
- **License Validation Logic**: Adds validation for free tier limits, license expiry, and node count limits
- **Periodic Validation**: Runs validation at startup and hourly in the orchestrator
- **User-Friendly Warnings**: Shows descriptive warnings to users when submitting jobs

## Example Warning
When exceeding free tier limits, users see:
```
Warning: Wow, you're popular! You've got 8 nodes, but the free tier only covers 5. Level up at https://cloud.expanso.io to connect your whole node family!
```

## Testing
- Unit tests cover all validation states and edge cases
- Tested warning display in client applications
- Verified license validation logic against test licenses

### 1. Valid License Message
**When**: Shown in logs when a license is valid and within limits
```
All good! Your license is valid and working its magic! 🎉 Keep on computing!
```

### 2. Free Tier Valid Message
**When**: Shown in logs and to users when no license is detected but node count is within free tier limits
```
No license? No problem! You can use Bacalhau for free with up to 5 nodes, or grab a license at https://cloud.expanso.io when you're ready to expand
```

### 3. Free Tier Exceeded Message
**When**: Shown in logs and to users when no license is detected and node count exceeds free tier limits
```
Wow, you're popular! You've got X nodes, but the free tier only covers 5. Level up at https://cloud.expanso.io to connect your whole node family!
```

### 4. Expired License Message
**When**: Shown in logs and to users when a license is expired
```
Oops! Your license expired on YYYY-MM-DD. You can renew it at https://cloud.expanso.io, or stick with the free tier (5 nodes max) for now
```

### 5. Exceeded Nodes Message
**When**: Shown in logs and to users when a license is valid but node count exceeds license limits
```
Your nodes are having quite the party! You have X nodes but your license covers Y. Upgrade your license at https://cloud.expanso.io for a bigger dance floor!
```

## Display Locations
- **Orchestrator Logs**: All messages are logged at startup and during hourly validation checks
- **Client Warnings**: Messages are displayed to users when submitting jobs

## Summary by CodeRabbit

- **New Features**
  - Updated job output so that warnings display only when full details are requested, with a new flag to hide warnings.
  - Enhanced licensing feedback with clear messages for valid, expired, or exceeded limits and an option to bypass validation via an environment variable.
  - Improved system status reporting by accurately tracking connected nodes.

- **Refactor**
  - Streamlined warning display and license handling logic to offer a cleaner, more intuitive user experience.

Closes ENG-500
Closes ENG-501

### CODE REVIEW

**Comment by wdbaruni** [comment]
@aronchick will appreciate your feedback here, specially on the message tone we are returning

File: pkg/licensing/constants.go [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+	// FreeTierMaxNodes is the maximum number of nodes allowed in the free tier
+	FreeTierMaxNodes = 5
+
+	// License validation messages
+	licenseMessageSkipped          = "License validation is taking a vacation today! 🏖️ (Validation is currently disabled)"
+	licenseMessageValid            = "All good! Your license is valid and working its magic! 🎉 Keep on computing!"
+	licenseMessageFreeTierValid    = "No license? No problem! You can use Bacalhau for free with up to %d nodes, or grab a license at https://cloud.expanso.io when you're ready to expand"
+	licenseMessageFreeTierExceeded = "Wow, you're popular! You've got %d nodes, but the free tier only covers %d. Level up at https://cloud.expanso.io to connect your whole node family!"
+	licenseMessageExpired          = "Oops! Your license expired on %s. You can renew it at https://cloud.expanso.io, or stick with the free tier (%d nodes max) for now"
+	licenseMessageExceededNodes    = "Your nodes are having quite the party! You have %d nodes but your license covers %d. Upgrade your license at https://cloud.expanso.io for a bigger dance floor!"
```
Comment (aronchick) [inline pkg/licensing/constants.go:19]: ...your license only covers...

**Review by aronchick — APPROVED** [review]
